### DISCUSSION TITLE
(pipelines): how to access jar files built in previous stages

### DISCUSSION THREAD

**Discussion by vimuston** [aws/aws-cdk#19713] [Q&A]
### Describe your issue?

I'm migrating to the new modern pipelines and CDK v2 and I'm having problems understanding how I could achieve the following things

Given:
* I have a repository with a Java/Spring Boot application, built with maven
* In this repository I have a Dockerfile that copies the built jar and runs it

I am using the typescript version of aws-cdk to build my pipeline. 

I'm trying to build the jar in my synth stage, along with security scans and Sonar analysis. This is no problem.

Later in the application stage I have 
`
  const image = ecs.ContainerImage.fromAsset(
    '../',
      file: 'docker/Dockerfile',
    },
  );
`

When the pipeline reaches the application stage, it can no longer find the jar file I built in the synth stage. I tried adding the maven target directory as an output to the synth stage, and that packages it as an output to S3. But it is not available in the assets stage so that I can build the actual container.

I have a feeling I'm approaching this the wrong way, but I can't find a simple example for a similar case anywhere.

**Comment by madeline-k** [comment]
I am pretty sure this is a guidance issue and not a feature request. @ryparker or @peterwoodworth can you provide some assistance here?
